iaso/tests/api/test_form_versions_readonly.py

```diff
@@ -0,0 +1,192 @@
+
+@override_settings(MEDIA_ROOT=tempfile.mkdtemp())
+class ReadOnlyFormsVersionAPITestCase(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        blue_account = m.Account.objects.create(name="Blue Account")
+        red_account = m.Account.objects.create(name="Red Account")
+
+        blue_source = m.DataSource.objects.create(name="Blue Source")
+        blue_source_version = m.SourceVersion.objects.create(data_source=blue_source, number=1)
+        blue_source.default_version = blue_source_version
+        blue_source.save()
+
+        cls.anon = AnonymousUser()
+
+        cls.blue_with_perms = cls.create_user_with_profile(
+            username="blue_with_perms", account=blue_account, permissions=["iaso_forms"]
+        )
+        cls.red_with_perms = cls.create_user_with_profile(
+            username="red_with_perms", account=red_account, permissions=["iaso_forms"]
+        )
+        cls.red_no_perms = cls.create_user_with_profile(username="red_no_perms", account=red_account)
+
+        cls.blue_council = m.OrgUnitType.objects.create(name="Blue Council", short_name="BC1")
+
+        cls.blue_project_need_auth = m.Project.objects.create(
+            name="Blue Project Needs Auth",
+            app_id="blue.project_need_auth",
+            account=blue_account,
+            needs_authentication=True,
+        )
+        cls.blue_project_need_auth.unit_types.add(cls.blue_council)
+
+        cls.blue_project_no_need_auth = m.Project.objects.create(
+            name="Blue Project No Needs Auth",
+            app_id="blue.project_no_need_auth",
+            account=blue_account,
+            needs_authentication=False,
+        )
+
+        cls.blue_project_no_need_auth.unit_types.add(cls.blue_council)
+        cls.blue_project_no_need_auth.save()
+
+        cls.form_need_auth = create_add_form("Form need_auth", cls.blue_council, cls.blue_project_need_auth)
+        cls.form_no_need_auth = create_add_form("Form no_need_auth", cls.blue_council, cls.blue_project_no_need_auth)
+
+    def test_form_versions_list_without_auth(self):
+        """GET /mobile/formversions/: without auth should return no versions"""
+
+        response = self.client.get(BASE_URL)
+        self.assertJSONResponse(response, 200)
+        response_data = response.json()
+        assert response_data["form_versions"] == []
+
+    def test_form_versions_list_without_auth_with_appid(self):
+        """GET /mobile/formversions/: without auth but with app_id should return the version of the form which allow access without auth"""
+
+        response = self.client.get(f"{BASE_URL}?app_id={self.blue_project_no_need_auth.app_id}")
+        self.assertJSONResponse(response, 200)
+        form_versions_data = response.json()["form_versions"]
+
+        assert len(form_versions_data) == 1
+
+        for form_version_data in form_versions_data:
+            self.assertValidFormVersionData(form_version_data)
+            self.assertNotIn("descriptor", form_version_data)
```

Comment:
Why do you test against that?
I need the descriptors when requested, so I want to ensure they'll still be provided, even for anonymous users, when requested.